src/Arch/X86/X86Disassembler.cs

```diff
@@ -34,21 +34,277 @@ namespace Reko.Arch.X86
 	/// </summary>
 	public partial class X86Disassembler : DisassemblerBase<X86Instruction>
 	{
+        private class X86LegacyCodeRegisterExtension
+        {
+            const byte MAGIC = 0x40;
+            const byte MAGIC_MASK = 0xf0;
+            internal static X86LegacyCodeRegisterExtension Disabled = new X86LegacyCodeRegisterExtension(0);
+            
+            byte val;
+
+            internal X86LegacyCodeRegisterExtension(byte value)
+            {
+                this.val = value;
+            }
+            internal X86LegacyCodeRegisterExtension(byte magic, bool wide, bool modrm_reg, bool sib_idx, bool modrm_rm)
+            {
+                this.val = (byte)((this.val & 0xf) | ((magic & 0xf) << 4));
+                this.WideValue = wide;
+                this.Target_ModRM_Reg = modrm_reg;
+                this.Target_SIB_Index = sib_idx;
+                this.Target_ModRM_RegOrMem = modrm_rm;
+            }
+
+            internal byte ByteValue
+            {
```

Comment:
For compactness I prefer that properties that don't raise events be automatic properties, ie
internal byte ByteValue {get; set; }